### ISSUE TITLE
[TF2; Feature Request] Auto-ban players from Casual matchmaking after being kicked too many times

### ISSUE THREAD

**Issue by tariqmudallal** [ValveSoftware/Source-1-Games#5290]
You know how we are kicking aimbotters from Casual servers? And then they find a way back in or different bots appear and then we have to do our due diligence on kicking them all? What if we made it so that if the bots get kicked enough times, say like 30 times at once, then they will get a matchmaking ban, something that could vary in length from a week to a month to a year or even a lifetime? Also, take into account how many recent reports the hacker got at once in the recent weeks or months to determine whether the player being kicked deserves a ban of such. We also need to patch up the vote-kicking system due to the fact when we try to vote out the bots they tend to still count as a "fail" since bots are trying to stay in the game and ruin the fun, why I said we should[ break up Casual lobbies every game](https://github.com/ValveSoftware/Source-1-Games/issues/3363). I think if we made it so that 11 votes end, the vote immediately ends and counts as a pass, which could be a much-needed change.

**Comment by Platina6978** [comment]
I agree that kicking cheaters isn't a particularly fun thing to do and TF2 needs an anti-cheat update most of all, given all the bots and cheaters just doing their thing for god knows how long now. Auto-banning people based on the amount of reports or kicks creates a system that is extremely abusable by bad actors. Just imagine people running bots systematically targeting innocent people with kicks and reports with their 10/20/30/... amount of bot accounts to use for this purpose.

**Comment by SomeBloodyMinger** [comment]
What I think they should do instead is: if the votekick is successful, then you get no cooldown. This will make speed-kicking boatloads of bots much less tedious. Cooldowns when teams are slowly getting rolled/ piled on by bots are a large potential risk for servers getting overpopulated and players having to abandon them. Players being able to kick immediately after can also free up the game much quicker to lower the likelihood of bots ruining the fun for enemy players before getting kicked. 

And if it's the bot that happens to get a kick, the cooldown won't matter much anyway because that server is already likely botrolled by that point. It'd be best to just leave servers that are full of too many bots.
